### PR TITLE
Modified users so that the attributes doesn't change when adding an authority.

### PR DESCRIPTION
**Pull request by Vim0315**
To do this, I added setter and getter for UUID in the Identifiable class. Is this okay?
The corresponding tests are added and passed.
I have also added some proper java doc to entity classes, more will be added in the future.
Sorry for the long pull request, it was too late when I realized.

### CODE REVIEW

File: src/test/java/Users/AdminTest.java [hunk 1]
```diff
@@ -44,4 +45,23 @@ public void testDoubleAuthority() {
...
+        User BothAdmin = new BanAuthority(VoteAdmin);
+        assert(BothAdmin.getUserPortfolio().equals(VoteAdmin.getUserPortfolio()));
+    }
+
+    @Test(timeout = 500)
+    public void testConstantName(){
+        User BothAdmin = new BanAuthority(VoteAdmin);
+        assert(BothAdmin.getName().equals("Leon"));
+    }
 }
```
Comment (tammyliuu) [inline src/test/java/Users/AdminTest.java:67]: Looking good so far.

**Comment by langsonzhang** [comment]
Looks good. But I have one question: why do we need to set() the UUID? I don't see a scenario where we would need to change the UUID of an object. Also, this causes a big problem since we hash using the UUID, and we rely on the fact that it does not change once initialized. So overall, I think we should keep UUID as a private final instance. Note that the UUID itself is an immutable class, so you cannot change it even if you have direct access using get().

File: src/main/java/Identification/Identifiable.java [hunk 2]
```diff
@@ -15,7 +15,7 @@
 
 public abstract class Identifiable {
 
-    public final UUID id;
+    public UUID id;
```
Comment (langsonzhang) [inline src/main/java/Identification/Identifiable.java:18]: Should be final. This allows misuse via hashing.

File: src/main/java/Identification/Identifiable.java [hunk 3]
```diff
@@ -40,11 +41,23 @@ public boolean equals(Object other) {
...
      */
     @Override
     public int hashCode() {
         return Objects.hash(this.id);
     }
 
+    /**
+     * Setter for the Identifiable object's UUID.
+     */
+    public void setId(UUID id) {
```
Comment (langsonzhang) [inline src/main/java/Identification/Identifiable.java:54]: Same idea. Changing the UUID will cause problems.

File: src/main/java/Identification/Identifiable.java [hunk 4]
```diff
@@ -15,7 +15,7 @@
 
 public abstract class Identifiable {
```
Comment (langsonzhang) [inline src/main/java/Identification/Identifiable.java:16]: getId() is a good addition. But I think others should not be added.

**Review by langsonzhang — CHANGES_REQUESTED** [review]

**Pull request closed without merging**